### PR TITLE
Remove some unused boxed closure code

### PR DESCRIPTION
**Pull request by arielb1**
It is not used anymore

### CODE REVIEW

File: src/librustc/middle/ty_fold.rs [hunk 1]
```diff
@@ -699,11 +689,9 @@ pub fn super_fold_closure_ty<'tcx, T: TypeFolder<'tcx>>(this: &mut T,
                                                         -> ty::ClosureTy<'tcx>
 {
     ty::ClosureTy {
-        store: fty.store.fold_with(this),
         sig: fty.sig.fold_with(this),
         unsafety: fty.unsafety,
-        onceness: fty.onceness,
-        bounds: fty.bounds.fold_with(this),
+//        bounds: fty.bounds.fold_with(this),
```
Comment (alexcrichton) [inline src/librustc/middle/ty_fold.rs]: Should this be removed entirely?

**Comment by alexcrichton** [comment]
Nice cleanup!

**Comment by arielb1** [comment]
Fixed nits.

**Pull request merged**